app/bundles/LeadBundle/Entity/LeadDeviceRepository.php

```diff
@@ -98,4 +98,31 @@ public function getDevice($lead, $deviceName = null, $deviceBrand = null, $devic
 
         return (!empty($device)) ? $device[0] : [];
     }
+
+    /**
+     * @param string $fingerprint
+     *
+     * @return LeadDevice
+     */
+    public function getDeviceByFingerprint($fingerprint)
+    {
+        if (!$fingerprint) {
+            return null;
+        }
+
+        $sq = $this->_em->getConnection()->createQueryBuilder();
+        $sq->select('es.id as id, es.lead_id as lead_id')
+            ->from(MAUTIC_TABLE_PREFIX.'lead_devices', 'es');
+
+        if ($fingerprint !== null) {
+            $sq->where(
+                $sq->expr()->eq('es.device_fingerprint', ':fingerprint')
+            )
+                ->setParameter('fingerprint', $fingerprint);
+        }
+        //get totals
+        $device = $sq->execute()->fetchAll();
+
+        return (!empty($device)) ? $device[0] : [];
```

Comment:
Since we work only with the first row, could we limit the whole query to fetch only 1 row? There is a possibility it will fetch much more and it will be just waste of resources.